token-swap/js/cli/token-swap-test.js

```diff
@@ -325,6 +328,74 @@ export async function withdraw(): Promise<void> {
   assert(info.amount.toNumber() == tokenB);
 }
 
+export async function swapOneTransaction(): Promise<void> {
+  console.log('Creating swap token a account');
+  let userAccountA = await mintA.createAccount(owner.publicKey);
+  await mintA.mintTo(userAccountA, owner, [], SWAP_AMOUNT_IN);
+
+  const balanceNeeded = await Token.getMinBalanceRentForExemptAccount(
+    connection,
+  );
+  const newAccount = new Account();
+  const transaction = new Transaction();
+  transaction.add(
+    SystemProgram.createAccount({
+      fromPubkey: mintB.payer.publicKey,
+      newAccountPubkey: newAccount.publicKey,
+      lamports: balanceNeeded,
+      space: AccountLayout.span,
+      programId: mintB.programId,
+    }),
+  );
+
+  transaction.add(
+    Token.createInitAccountInstruction(
+      mintB.programId,
+      mintB.publicKey,
+      newAccount.publicKey,
+      owner.publicKey,
+    ),
+  );
+
+  transaction.add(
+    Token.createApproveInstruction(
+      mintA.programId,
+      userAccountA,
+      authority,
+      owner.publicKey,
+      [owner],
+      SWAP_AMOUNT_IN,
+    )
+  );
+
+  transaction.add(
+    TokenSwap.swapInstruction(
+      tokenSwap.tokenSwap,
+      authority,
+      userAccountA,
+      tokenAccountA,
+      tokenAccountB,
+      newAccount.publicKey,
+      tokenSwap.programId,
+      mintB.programId,
+      SWAP_AMOUNT_IN,
+      SWAP_AMOUNT_OUT,
+    )
+  );
+
+  // Send the instructions
+  console.log('sending big instruction');
+  await sendAndConfirmTransaction(
+    'do a whole lot',
+    connection,
+    transaction,
+    owner,
+    newAccount,
+    mintA.payer,
+    mintB.payer,
```

Comment:
Does this transaction require 4 signatures? I'm not following why all these signers are required.

Reply:
You're absolutely right, I messed up one the first instruction which makes this require `mintB.payer` -- lemme get an update to you